subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -47,64 +47,154 @@
 static int init_reset(void);
 static void ticker_cb(u32_t ticks_at_expire, u32_t remainder, u16_t lazy,
 		      void *param);
-static u8_t disable(u16_t handle);
+static u8_t disable(u8_t handle);
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+#define BT_CTLR_SCAN_MAX 2
+#else /* !CONFIG_BT_CTLR_ADV_EXT */
 #define BT_CTLR_SCAN_MAX 1
+#endif /* !CONFIG_BT_CTLR_ADV_EXT */
+
 static struct ll_scan_set ll_scan[BT_CTLR_SCAN_MAX];
 
 u8_t ll_scan_params_set(u8_t type, u16_t interval, u16_t window,
 			u8_t own_addr_type, u8_t filter_policy)
 {
 	struct ll_scan_set *scan;
+	struct lll_scan *lll;
 
-	scan = ull_scan_is_disabled_get(0);
+	scan = ull_scan_is_disabled_get(SCAN_HANDLE_1M);
 	if (!scan) {
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	struct ll_scan_set *scan_coded;
+	u8_t phy;
+
+	scan_coded = ull_scan_is_disabled_get(SCAN_HANDLE_PHY_CODED);
+	if (!scan_coded) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	phy  = type >> 1;
+	if (phy & BT_HCI_LE_EXT_SCAN_PHY_CODED) {
+		scan = scan_coded;
+	}
+
+	lll = &scan->lll;
+
+	if (!interval) {
+		lll->phy = 0U;
+
+		return 0;
+	}
+
+	lll->phy = phy;
+
+#else /* !CONFIG_BT_CTLR_ADV_EXT */
+	lll = &scan->lll;
+#endif /* !CONFIG_BT_CTLR_ADV_EXT */
+
 	scan->own_addr_type = own_addr_type;
 
-	ull_scan_params_set(&scan->lll, type, interval, window, filter_policy);
+	ull_scan_params_set(lll, type, interval, window, filter_policy);
 
 	return 0;
 }
 
 u8_t ll_scan_enable(u8_t enable)
 {
+	struct ll_scan_set *scan_coded = NULL;
 	struct ll_scan_set *scan;
+	u8_t own_addr_type = 0U;
+	u8_t is_coded_phy = 0U;
+	u8_t err;
 
 	if (!enable) {
-		return disable(0);
+		err = disable(SCAN_HANDLE_1M);
+
+		if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT)) {
+			u8_t err_coded;
+
+			err_coded = disable(SCAN_HANDLE_PHY_CODED);
+			if (!err_coded) {
+				err = 0U;
+			}
+		}
+
+		return err;
 	}
 
-	scan = ull_scan_is_disabled_get(0);
+	scan = ull_scan_is_disabled_get(SCAN_HANDLE_1M);
 	if (!scan) {
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
-	if (scan->own_addr_type & 0x1) {
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	scan_coded = ull_scan_is_disabled_get(SCAN_HANDLE_PHY_CODED);
+	if (!scan_coded) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	own_addr_type = scan_coded->own_addr_type;
+	is_coded_phy = (scan_coded->lll.phy &
+			BT_HCI_LE_EXT_SCAN_PHY_CODED);
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
+
+	if ((IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT) && is_coded_phy &&
+	     (own_addr_type & 0x1)) ||
+	    (scan->own_addr_type & 0x1)) {
 		if (!mem_nz(ll_addr_get(1, NULL), BDADDR_SIZE)) {
 			return BT_HCI_ERR_INVALID_PARAM;
 		}
 	}
 
 #if defined(CONFIG_BT_CTLR_PRIVACY)
-	struct lll_scan *lll = &scan->lll;
+	struct lll_scan *lll;
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT) && is_coded_phy) {
+		lll = &scan_coded->lll;
```

Comment:
It looks like this code would update the state in only one of the lll_scans, not both. Maybe factoring out the relevant code into a function and calling it for both lll_scans would be appropriate?

Reply:
The privacy support is TODO, will need to remember to address then.